### PR TITLE
Codegen pow.Tensor_Scalar and pow.Tensor_Tensor

### PR DESCRIPTION
**Pull request by desertfire**
Summary: Codegen pow, and then remove ops.h|cpp

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h [hunk 1]
```diff
@@ -33,6 +33,38 @@ class Generic : public torch::lazy::TsNode {
   torch::lazy::hash_t hash_seed_;
 };
 
+inline torch::lazy::NodePtr GenericOp(
```
Comment (alanwaketan) [inline lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: Do we need all of the helpers? And you probably don't need inline as the function is defined in the header.

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h [hunk 2]
```diff
@@ -31,6 +31,16 @@ torch::lazy::hash_t ScalarHash(const at::Scalar& s);
 
 using at::operator<<;
 
+inline torch::lazy::NodePtr ScalarOp(const at::Scalar& value,
```
Comment (alanwaketan) [inline lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h:34]: Do we need all of the helpers? And you probably don't need inline as the function is defined in the header.

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h (hunk 1 above)
Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: which helpers? you mean do we need GenericOp anymore?

Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: We have 4 overloads here. I'm not sure if we actually need all of them.

Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: Initially PyTorch/XLA uses all the helpers here, but you are right, after switching to codegen, we don't need all of them.

Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: Not sure I understand your comment about`inline`. Being defined in the header doesn't guarantee a function will be inlined, right?

Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: There is no guaranteed `inline` as far as you know. The compiler has the free will to make the final call. Implicit `inline` in headers should be enough.

Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: I see `inline` is used in core almost as a coding convention, so I guess I would prefer to leave it as is.

Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/generic.h:36]: Hmm, up to you. We don't have any coding styles for reference anyway. I'm not going to force you. 😃

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Review by wconstab — APPROVED** [review]
thanks Bin, sorry I missed this, i do remember I started to review it and got interrupted.

**Pull request merged**